lib/Dialect/FIRRTL/Transforms/GrandCentral.cpp

```diff
@@ -1431,31 +1466,16 @@ Optional<TypeSum> GrandCentralPass::computeField(Attribute field,
 /// of the AugmentedBundleType to construct nested interfaces and generate
 /// stringy-typed SystemVerilog hierarchical references to drive the
 /// interface. Returns false on any failure and true on success.
-Optional<sv::InterfaceOp>
-GrandCentralPass::traverseBundle(AugmentedBundleTypeAttr bundle, IntegerAttr id,
-                                 StringAttr prefix, VerbatimBuilder &path) {
-  auto builder = OpBuilder::atBlockEnd(getOperation().getBodyBlock());
-  sv::InterfaceOp iface;
-  builder.setInsertionPointToEnd(getOperation().getBodyBlock());
-  auto loc = getOperation().getLoc();
-  auto iFaceName = getNamespace().newName(getInterfaceName(prefix, bundle));
-  iface = builder.create<sv::InterfaceOp>(loc, iFaceName);
-  ++numInterfaces;
-  if (dut &&
-      !instancePaths->instanceGraph.isAncestor(companionIDMap[id].companion,
-                                               cast<hw::HWModuleLike>(*dut)) &&
-      testbenchDir)
-    iface->setAttr("output_file", hw::OutputFileAttr::getAsDirectory(
-                                      &getContext(), testbenchDir.getValue(),
-                                      /*excludeFromFileList=*/true));
-  else if (maybeExtractInfo)
-    iface->setAttr("output_file",
-                   hw::OutputFileAttr::getAsDirectory(
-                       &getContext(), getOutputDirectory().getValue(),
-                       /*excludeFromFileList=*/true));
-  iface.setCommentAttr(builder.getStringAttr("VCS coverage exclude_file"));
-
-  builder.setInsertionPointToEnd(cast<sv::InterfaceOp>(iface).getBodyBlock());
+Optional<StringAttr> GrandCentralPass::traverseBundle(
+    AugmentedBundleTypeAttr bundle, IntegerAttr id, StringAttr prefix,
+    VerbatimBuilder &path, SmallVector<VerbatimXMRbuilder> &xmrElems,
+    SmallVector<InterfaceElemsBuilder> &interfaceBuilder) {
+
+  unsigned lastIndex = interfaceBuilder.size();
+  InterfaceElemsBuilder *ifaceBuilder = &interfaceBuilder.emplace_back();
```

Comment:
This will cause memory corruption when interfaceBuilder reallocates memory.

Reply:
Thanks @uenoku  for catching this. Fix in https://github.com/llvm/circt/pull/4351